server/src/main/java/org/apache/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -175,10 +174,20 @@ public File getSegmentFiles(DataSegment segment) throws SegmentLoadingException
    * location may fail because of IO failure, most likely in two cases:<p>
    * 1. druid don't have the write access to this location, most likely the administrator doesn't config it correctly<p>
    * 2. disk failure, druid can't read/write to this disk anymore
+   *
+   * Locations are fetched using {@link StorageLocationSelectorStrategy}.
    */
   private StorageLocation loadSegmentWithRetry(DataSegment segment, String storageDirStr) throws SegmentLoadingException
   {
-    for (StorageLocation loc : locations) {
+    Iterator<StorageLocation> locationsIterator = strategy.getLocations();
+    int numLocationsToTry = this.locations.size();
```

Comment:
`numLocationsToTry` is not necessary now.

Reply:
Oops! will fix it.

Reply:
https://github.com/apache/incubator-druid/pull/8038#discussion_r329008898 javadocs u mean ?

Reply:
Removed numLocationsToTry and update java docs. Let me know if the description isn't clear or any change is required.